packages/search-input/src/SearchInput/SearchInput.spec.tsx

```diff
@@ -0,0 +1,509 @@
+import React from 'react';
+import {
+  createEvent,
+  fireEvent,
+  waitForElementToBeRemoved,
+} from '@testing-library/dom';
+import userEvent from '@testing-library/user-event';
+import { axe } from 'jest-axe';
+
+import * as _LGTL from '@leafygreen-ui/testing-lib';
+
+import { renderSearchInput } from '../utils/SearchInput.testutils';
+import { SearchInput, SearchResult, Size } from '..';
+
+import { State } from './SearchInput.types';
+
+const resultClickHandler = jest.fn();
+
+const defaultProps = {
+  className: 'test-text-input-class',
+  placeholder: 'This is some placeholder text',
+  onChange: jest.fn(),
+  onClear: jest.fn(),
+  onBlur: jest.fn(),
+  children: [
+    <SearchResult key="a" onClick={resultClickHandler}>
+      Apple
+    </SearchResult>,
+    <SearchResult key="b" onClick={resultClickHandler}>
+      Banana
+    </SearchResult>,
+    <SearchResult key="c" onClick={resultClickHandler}>
+      Carrot
+    </SearchResult>,
+    <SearchResult key="d" onClick={resultClickHandler}>
+      Dragonfruit
+    </SearchResult>,
+  ],
+};
+
+describe('packages/search-input', () => {
+  describe('a11y', () => {
+    test('does not have basic accessibility issues', async () => {
+      const { container } = renderSearchInput();
+      const results = await axe(container);
+      expect(results).toHaveNoViolations();
+    });
+  });
+
+  describe('Basic rendering', () => {
+    test('renders type as "search"', () => {
+      const { inputEl } = renderSearchInput();
+      expect(inputEl.getAttribute('type')).toBe('search');
+    });
+
+    test(`renders provided placeholder text`, () => {
+      const { getByPlaceholderText } = renderSearchInput(defaultProps);
+      expect(getByPlaceholderText(defaultProps.placeholder)).toBeVisible();
+    });
+
+    test(`passes className to root element`, () => {
+      const { containerEl } = renderSearchInput(defaultProps);
+      expect(
+        containerEl.classList.contains(defaultProps.className),
+      ).toBeTruthy();
+    });
+
+    test('clear button is not rendered when there is no text', () => {
+      const { queryByRole } = renderSearchInput();
+      expect(queryByRole('button')).not.toBeInTheDocument();
+    });
+
+    test('clear button is rendered when there is text', () => {
+      const { queryByRole, inputEl } = renderSearchInput();
+      userEvent.type(inputEl, 'abc');
+      expect(queryByRole('button')).toBeInTheDocument();
+    });
+
+    describe('when the "size" is not "large"', () => {
```

Comment:
nit but why keep these in if they aren't doing anything?

Reply:
They're still testing for font size. I can remove them though since they'll be replaced by Chromatic soon enough